packages/react-server/src/ReactFizzServer.js

```diff
@@ -498,24 +498,28 @@ function renderSuspenseBoundary(
   boundarySegment.status = COMPLETED;
   boundarySegment.children.push(innerSegment);
 
-  // We create suspended task for the fallback because we don't want to actually work
-  // on it yet in case we finish the main content, so we queue for later.
-  const suspendedFallbackTask = createTask(
-    request,
-    fallback,
-    parentBoundary,
-    innerSegment,
-    fallbackAbortSet,
-    task.legacyContext,
-    task.context,
-    null,
-  );
-  if (__DEV__) {
-    suspendedFallbackTask.componentStack = task.componentStack;
+  // if avoidThisFallback is set to true then we act as though there is no fallback in SSR
+  // Instead, we let the client render the fallback if the component suspends there.
```

Comment:
We don’t actually want the client to render the fallback - unless there is an update that forces it to render. Even that is an unusual case. In the regular flow it would just show the parent fallback until it unsuspends.

Reply:
> unless there is an update that forces it to render.

Right, thats what I meant by letting the client render the fallback if the component suspends (eg. if the comments ranking is changed then it would suspend)